### PR TITLE
feat: 11642 app shimmer video

### PR DESCRIPTION
**Pull request by vikcodes**
## What does this PR do?

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by joeauyeung — DISMISSED** [review]
Tested and working on Cal.com side

File: packages/app-store/shimmervideo/config.json
Comment (hariombalhara) [inline packages/app-store/shimmervideo/config.json:1]: Removed `dirName` as it's legacy. `dirName` is considered same as `slug` in that case for legacy code using dirName.

Also, modified slug to be as `shimmervideo` which is same as the actual dir name.

Comment (hariombalhara) [reply packages/app-store/shimmervideo/config.json:1]: cc @joeauyeung

File: packages/core/EventManager.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ export default class EventManager {
     // (type closecom_other_calendar)
     this.calendarCredentials = appCredentials.filter((cred) => cred.type.endsWith("_calendar"));
     this.videoCredentials = appCredentials
-      .filter((cred) => cred.type.endsWith("_video"))
+      .filter((cred) => cred.type.endsWith("_video") || cred.type.endsWith("_conferencing"))
```
Comment (hariombalhara) [inline packages/core/EventManager.ts:100]: I am not sure why it's needed @joeauyeung as we are still using _video postfix for conferencing apps and using the same for shimmer app too . Leaving it as is for now.

File: packages/app-store/shimmervideo/lib/VideoApiAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+export interface DailyVideoCallData {
+  type: string;
+  id: string;
+  password: string;
+  url: string;
+}
+
+export const fetcher = async (endpoint: string, init?: RequestInit | undefined) => {
+  const { api_key } = await getShimmerAppKeys();
+  const response = await fetch(`https://api.daily.co/v1${endpoint}`, {
```
Comment (hariombalhara) [inline packages/app-store/shimmervideo/lib/VideoApiAdapter.ts:62]: Why are we hitting daily API from shimmer video app. If shimmer video app uses daily itself, it shouldn't depend on we hitting daily API. We should contact shimmer app only for that. Shimmer app can do that same at it's backend

**Comment by hariombalhara** [comment]
@vikcodes Now env variables are removed and configurable from UI. See https://www.loom.com/share/77591402656d4aa2a15fe76bb0ae2a3a

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks @vikcodes. I have one concern before I can merge it.

I would not recommend hitting daily endpoint from shimmer app. Let's proxy that endpoint from some shimmer endpoint only

**Review by joeauyeung — APPROVED** [review]
Talking with the team, we're going to approve now to get it on production and then refactor at a later date.

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**